### PR TITLE
Start on Episode form

### LINKED ISSUE #534: Edit Episode
After #505.  Start building out the [Edit Episode](https://www.figma.com/proto/Yewv83Q8btHcaOBWCtGgKs/PRX-Dovetail-with-Navigation?page-id=80%3A2947&node-id=2375%3A23125&viewport=798%2C-3032%2C0.25&scaling=min-zoom&starting-point-node-id=80%3A3420&hide-ui=1) form.  Goal here is to get things stubbed in, and leave the more complex parts for separate tickets.

image

- [ ] Setup the routes/tabs. Use `episodes#edit` and `/episodes/<guid>/edit`
  - (NOTE: I'm sort of reserving `/episodes/<guid>` show for something else in the future.  Because I think it's just a matter of time before we'll want that).
- [ ] Build out the page layout in bootstrap - just stub in all the panels for now.
- [ ] Implement just the fields highlighted in the screenshot, and leave the WYSIWYG as plain textareas for now
- [ ] Use some turbo or stimulus for the alternate itunes summary
- [ ] Add the save button, but leave the status box / publishing stuff out for now.

### PR DESCRIPTION
**Pull request by cavis**
Closes #534.  Will open additional tasks for remaining parts of this (audio, prefill-tags, images, etc etc).

### CODE REVIEW

File: app/assets/stylesheets/shared/form.scss [hunk 1]
```diff
@@ -39,6 +39,25 @@
   }
 }
 
+// bootstrap5-tags need help
```
Comment (cavis) [inline app/assets/stylesheets/shared/form.scss:42]: Found/added this [bootstrap5-tags](https://github.com/lekoala/bootstrap5-tags) library.  Mostly works and we can add autocomplete ... but needed (needs?) some styling help.

File: app/helpers/episodes_helper.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+require "text_sanitizer"
+
+module EpisodesHelper
+  def episode_itunes_type_options
+    Episode::VALID_ITUNES_TYPES.map { |val| [I18n.t("helpers.label.episode.itunes_types.#{val}"), val] }
+  end
+
+  def episode_status_class(episode)
+    case episode.publishing_status_was
```
Comment (cavis) [inline app/helpers/episodes_helper.rb:9]: Bit tricky ... but I needed the color of the status box to match the _previous_ publishing_status, when you submit something invalid.

File: app/helpers/feeder_form_builder.rb [hunk 3]
```diff
@@ -50,14 +53,7 @@ def date_field(method, options = {})
 
   def time_field(method, options = {})
     value = options[:value] || object&.public_send(method)
-
-    # end timestamps are fudged 1-minute to look inclusive
-    if options[:fudge] && value && value == value.beginning_of_day
```
Comment (cavis) [inline app/helpers/feeder_form_builder.rb:55]: Removed all the code around "fudging" timestamps by 1 second.  That was an Augury end_date thing.

File: app/models/concerns/publishing_status.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      "draft"
+    elsif published_at_was > Time.now
+      "scheduled"
+    else
+      "published"
+    end
+  end
+
+  def publishing_status=(value)
+    @publishing_status = value
```
Comment (cavis) [inline app/models/concerns/publishing_status.rb:34]: Adds getters/setters for the `publishing_status` virtual field.  Which allows the UX for the status-changer.

File: app/models/episode.rb [hunk 5]
```diff
@@ -79,6 +82,11 @@ def self.story_uri(story)
     (story.links["self"].href || "").gsub("/authorization/", "/")
   end
 
+  # use guid rather than id for episode routes
+  def to_param
+    guid
```
Comment (cavis) [inline app/models/episode.rb:87]: Bad idea to use guids in the routes, vs the database ids?

File: app/models/podcast.rb [hunk 6]
```diff
@@ -26,7 +26,7 @@ class Podcast < ApplicationRecord
   # these keep changing - so just translate to the current accepted values
   VALID_EXPLICITS = %w[true false]
   EXPLICIT_ALIASES = {
-    "" => "false",
+    "" => nil,
```
Comment (cavis) [inline app/models/podcast.rb:29]: This seemed like an oversight in Publish ... you should be able to set this to `nil` on Episodes.

**Review by brandonhundt — APPROVED** [review]
Good start, LGTM

**Pull request merged**